### PR TITLE
fix: smart count type

### PR DESCRIPTION
**Pull request by davidfurlong**
Unfortunately having one optional property a different type in an object is very difficult to do in typescript and requires weird object instantiation when it is possible. So will have to sacrifice type safety here a bit

======================================
Running tests for locale en
======================================
✅ Passed test 1, en
✅ Passed test 2, {Hello, Test.}
✅ Passed test 3, {Hello, Lowercase.}
✅ Passed test 4, {10 cars}
✅ Passed test 5, {10 cars}
✅ Passed test 6, {English}
✅ Passed test 7, {German}
✅ Passed test 8, {Select a language}
✅ Passed test 9, {English, German}
✅ Passed test 10, {MissingKey: missing-key}
✅ Passed test 11, {.preprocess.}
✅ Passed test 12, {David}
✅ Passed test 13, {undefined}
✅ Passed test 14, {MissingKey: missing-key}
✅ Passed test 15, {Global}
✅ Passed test 16, {MissingKey: english_lower}
======================================
✅ All tests passed!

======================================
Running tests for locale de
======================================
✅ Passed test 1, de
✅ Passed test 2, {Hallo, Test}
✅ Passed test 3, {Hallo, Lowercase}
✅ Passed test 4, {10 autos}
✅ Passed test 5, {10 autos}
✅ Passed test 6, {Englisch}
✅ Passed test 7, {Deutsch}
✅ Passed test 8, {Bitte Sprache wählen}
✅ Passed test 9, {Englisch, Deutsch}
✅ Passed test 10, {MissingKey: missing-key}
✅ Passed test 11, {.preprocess.}
✅ Passed test 12, {David}
✅ Passed test 13, {undefined}
✅ Passed test 14, {MissingKey: missing-key}
✅ Passed test 15, {Global}
✅ Passed test 16, {MissingKey: english_lower}
======================================
✅ All tests passed!

### CODE REVIEW

File: src/index.ts [hunk 1]
```diff
@@ -53,7 +53,8 @@ export type TranslateGlobal<
 > = TranslateType<L, TG>;
 
 export type Interpolation =
-  | ({ smart_count?: number } & {
-      [key: string]: string | object;
-    })
+  | {
```
Comment (michalkvasnicak) [inline src/index.ts:57]: isn't this possible to type like this?

```ts
type Interpolation = {
    smart_count?: number;
    [key: string]: string | number | object | undefined; 
};
```

https://www.typescriptlang.org/play?#code/C4TwDgpgBAkgdsCAnMB7ANgQ2AS1XKAXigG8AoKSqAZwFtMlgB9AY1QFcEB+ALijna0ARsgDcFKgG0A1hBB9qwJDjgBzALoKlK1VAA+-QSKT6oqIQCsILYKc4ATCADMVEe6KhkAvqKA

**Review by michalkvasnicak — CHANGES_REQUESTED** [review]

File: src/index.ts (hunk 1 above)
Comment (davidfurlong) [reply src/index.ts:57]: you are right that is better

Comment (davidfurlong) [reply src/index.ts:57]: I was only trying to get the _correct_ solution to work which meant keys that aren't `smart_count` shouldn't be number

Comment (michalkvasnicak) [reply src/index.ts:57]: @davidfurlong yes but numbers in translations should be allowed, right?

Comment (davidfurlong) [reply src/index.ts:57]: Improved, thanks

**Review by michalkvasnicak — APPROVED** [review]

**Pull request merged**